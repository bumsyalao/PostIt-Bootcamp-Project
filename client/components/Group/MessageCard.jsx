import React from 'react';

/**
 *
 *
 * @param {any} props 
 * @returns 
 */
const MessageCard = (props) => {
  const date = new Date(props.createdAt);
  return (
    <li className="collection-item">
      <div className="row">
        <div className="col s12 container">
          <div className="message-card">
            <div className="message-head">
              <div className="sender">
                <h6>{(props.username || props.User.username).toUpperCase()}</h6>
              </div>
              <div className="priority">
                Priority: &nbsp;
                <span>{props.messagePriority}</span>
              </div>
              <div className="time">
              {
                `${date.toDateString()} at ${
                  date.getHours()}:${date.getMinutes()}`
              }
              </div>
            </div>
            <div className="message-body"> {props.message}</div>
          </div>
        </div>
      </div>
    </li>
  );
};
export default MessageCard;
