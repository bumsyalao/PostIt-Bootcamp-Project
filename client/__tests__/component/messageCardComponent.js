/* global expect jest test */
import React from 'react';
import { shallow, configure } from 'enzyme';
import toJson from 'enzyme-to-json';
import adapter from 'enzyme-adapter-react-16';
import MessageCard from '../../components/Group/MessageCard';

configure({ adapter: new adapter() });

describe('MessageCard Component', () => {
  const spy = jest.fn();
  beforeEach(() => {
    global.Materialize = { toast: jest.fn() };
  });
  afterEach(() => {
    spy.mockReset();
  });
  const props = {
    createdAt: spy,
    username: 'blessing',
    User: { username: 'blessing' },
    messagePriority: spy,
    message: spy,

  };

  const component = shallow(
    <MessageCard {...props} />
  );

  it('should render component', () => {
    expect(component.exists()).toBe(true);
  });

  it('should match the Message card snapshot', () => {
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });
});
