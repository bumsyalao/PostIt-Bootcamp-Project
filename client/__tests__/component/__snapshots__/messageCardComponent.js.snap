// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageCard Component should match the Message card snapshot 1`] = `
<div
  className="row"
>
  <div
    className="col s12"
  >
    <div
      className="card white"
    >
      <div
        className="card-content black-text"
      >
        <span
          className="card-title inline-block"
        >
          @
          BLESSING
        </span>
        <span
          className="inline-margin inline-block priority-badge black-text"
        />
        <span
          className="inline-margin inline-block right"
        >
          Invalid Date at NaN:NaN
        </span>
      </div>
      <div
        className="card-action"
      >
        <p
          className="message-text"
        >
           
        </p>
      </div>
    </div>
  </div>
</div>
`;
